csrc/ops/alias.cpp

```diff
@@ -704,36 +701,58 @@ TensorView* slice(TensorView* inp, const std::vector<Slice>& ranges) {
       ", Expected: ",
       ndims);
 
-  auto normalize_slice_range = [](Slice range, Val* extent) -> Slice {
+  const auto normalize_slice_range = [](Slice range, Val* extent) -> Slice {
+    auto cast_extent =
+        SimplifyingIrBuilder::maybeCastExpr(DataType::Index, extent);
+
+    auto zero = FusionGuard::getCurFusion()->zeroVal(DataType::Index);
+
+    // norm_start = max(0, start < 0 ? start + extent : start)
```

Comment:
What if start > extent? In python, this is valid:
```python
[1,2,4][10000:]
```
is
```python
[]
```

Reply:
Oh, sorry, looks like this case is already tested.

Reply:
Yes, I wrote it this way to avoid an additional ternary expression, instead of clipping start to `extent`. If `start > extent`, then `stop` will get clipped to `start`, since it is first clipped to `extent` then to `start`.